### ISSUE TITLE
[BUG] Silence marker 'sil' is replaced by 'spn' while alignment

### ISSUE THREAD

**Issue by stefantaubert** [MontrealCorpusTools/Montreal-Forced-Aligner#392]
**Debugging checklist**

[x] Have you updated to latest MFA version? -> 2.0.0rc2
[x] Have you tried rerunning the command with the `--clean` flag?

**Describe the issue**
In the pronunciation dictionary I assigned to word `--` the silence marker `sil` (as described [here](https://montreal-forced-aligner.readthedocs.io/en/latest/user_guide/formats/dictionary.html?highlight=sil#non-speech-annotations)) but after the alignment `spn` is assigned instead of nothing because `sil` was considered as a missing phone.

**For Reproducing your issue**

1. Corpus structure
   * What language is the corpus in? -> English
   * How many files/speakers? -> One file / one speaker
   * Are you using lab files or TextGrid files for input? -> TextGrid files with one tier containing the whole text
2. Dictionary
   * Are you using a dictionary from MFA? If so, which one? -> No, custom dictionary containing all words separated by space
   * If it's a custom dictionary, what is the phoneset? -> ARPAbet
3. Acoustic model
   * If you're using an acoustic model, is it one download through MFA? If so, which one? -> Yes, the `english` model
   * If it's a model you've trained, what data was it trained on? -> /

command:

```
mfa align \
  "~/mfa/clean_grids" \
  "~/mfa/align.dict" \
  english \
  "~/mfa/clean_grids_aligned" \
  --config_path="~/mfa/align.conf" \
  --audio_directory="~/mfa/align_audio" \
  --temp_directory="~/mfa/MFA_tmp" \
  --num_jobs=16 \
  --verbose \
  --clean \
  --overwrite
```

excerpt from `align.dict`:

```
...
(WHICH  HH W IH1 CH
--  sil
A  AH0
A-ROTTING  AH0 R AA1 T IH0 NG
...
```

`align.conf`:

```
beam: 200
retry_beam: 400
transition_scale: 1.0
acoustic_scale: 0.1
self_loop_scale: 0.1
boost_silence: 1.0
punctuation: "♪"
clitic_markers: "♪"
compound_markers: "♪"
strip_diacritics: "♪"
digraphs: "♪"
```

Note: I've used `♪` to ignore all of the MFA dictionary lookup functionality because all words (incl. special chars) are in `align.dict`.

**Log**

`mfa validate ...` output:

```
...
DEBUG - ====ACOUSTIC MODEL INFO====
DEBUG - Acoustic model root directory: ~/Documents/MFA/extracted_models
DEBUG - Acoustic model dirname: ~/Documents/MFA/extracted_models/english
DEBUG - Acoustic model meta path: ~/Documents/MFA/extracted_models/english/meta.yaml
DEBUG - Acoustic model meta information:
DEBUG - URL: https://montrealcorpustools.github.io/Montreal-Forced-Aligner/ architecture: gmm-hmm authors: Michael
                 McAuliffe citation: 'M. McAuliffe, M. Socolof, S. Mihuc, M. Wagner, and M. Sonderegger (2017).
                 Montreal Forced Aligner: an accurate and trainable aligner using Kaldi. Presented   at the 91st Annual
                 Meeting of the Linguistic Society of America, Austin, TX.' features:   allow_downsample: true
                 allow_upsample: true   feature_type: mfcc   fmllr_update_type: full   frame_shift: 10   high_frequency:
                 7800   low_frequency: 20   pitch: false   sample_frequency: 16000   silence_weight: 0.0   snip_edges:
                 true   splice_left_context: 3   splice_right_context: 3   use_energy: false   uses_cmvn: true
                 uses_deltas: true   uses_speaker_adaptation: false   uses_splices: false   uses_voiced: false language:
                 English oov_phone: spn optional_silence_phone: sil other_noise_phone: sp phone_set_type: UNKNOWN
                 phone_type: triphone phones: !!set   AA0: null   AA1: null   AA2: null   AE0: null   AE1: null   AE2:
                 null   AH0: null   AH1: null   AH2: null   AO0: null   AO1: null   AO2: null   AW0: null   AW1: null
                 AW2: null   AY0: null   AY1: null   AY2: null   B: null   CH: null   D: null   DH: null   EH0: null
                 EH1: null   EH2: null   ER0: null   ER1: null   ER2: null   EY0: null   EY1: null   EY2: null   F: null
                 G: null   HH: null   IH0: null   IH1: null   IH2: null   IY0: null   IY1: null   IY2: null   JH: null
                 K: null   L: null   M: null   N: null   NG: null   OW0: null   OW1: null   OW2: null   OY0: null   OY1:
                 null   OY2: null   P: null   R: null   S: null   SH: null   T: null   TH: null   UH0: null   UH1: null
                 UH2: null   UW0: null   UW1: null   UW2: null   V: null   W: null   Y: null   Z: null   ZH: null
                 version: 0.9.0
...
  Sound file read errors
  ======================

    There were no issues reading sound files.

  Feature generation
  ==================

    There were no utterances missing features.

  Files without transcriptions
  ============================

    There were no sound files missing transcriptions.

  Transcriptions without sound files
  ==================================

    There were no transcription files missing sound files.

  TextGrid read errors
  ====================

    There were no issues reading TextGrids.


**********
Dictionary
**********

  Out of vocabulary words
  =======================

    1 OOV word types
    26 total OOV tokens

    For a full list of the word types, please see:

        ~/mfa/MFA_tmp/clean_grids_validate_pretrained/oovs_found.txt

    For a by-utterance breakdown of missing words, see:

        ~/mfa/MFA_tmp/clean_grids_validate_pretrained/utterance_oovs.txt


  Acoustic model compatibility
  ============================

    1 phones not in acoustic model
    1 ignored pronunciations

    Phones missing acoustic models:

        sil

...
```

**Desktop**
 - OS: [e.g. Windows, OSX, Linux] -> Linux
 - Version [e.g. MacOSX 10.15, Ubuntu 20.04, Windows 10, etc] -> Ubuntu 20.04.3 LTS (GNU/Linux 5.9.0-050900-generic x86_64)
 - Any other details about the setup (Cloud, Docker, etc) -> /

**Context**

Example sentence that should be aligned: `What little supper we were able to get on board was worse than none at all, for it did not stay with us -- anything but fun, this going to sea in a bowl, to rob gull's nests, and smuggle eggs into market.`

**Comment by stefantaubert** [comment]
I've just found out that `spn` will be considered as 'OOV word type', too.

```
  Acoustic model compatibility
  ============================

    1 phones not in acoustic model
    1 ignored pronunciations

    Phones missing acoustic models:

        spn

```

**Comment by mmcauliffe** [comment]
Should be fixed in 2.0.0rc4.
